validator/src/main/java/com/google/daq/mqtt/sequencer/sequences/PointsetSequences.java

```diff
@@ -1,81 +1,169 @@
 import static udmi.schema.FeatureEnumeration.FeatureStage.BETA;
 
 import com.google.daq.mqtt.sequencer.Feature;
-import com.google.daq.mqtt.sequencer.SequenceBase;
+import com.google.daq.mqtt.sequencer.PointsetBase;
 import com.google.daq.mqtt.sequencer.Summary;
 import com.google.daq.mqtt.util.SamplingRange;
 import java.time.Instant;
 import java.util.ArrayList;
 import java.util.Collections;
 import java.util.Date;
 import java.util.List;
+import java.util.Map.Entry;
 import java.util.stream.Collectors;
-import org.junit.AssumptionViolatedException;
 import org.junit.Test;
+import udmi.schema.Level;
+import udmi.schema.PointPointsetConfig;
+import udmi.schema.PointPointsetEvent;
+import udmi.schema.PointPointsetState;
 import udmi.schema.PointsetEvent;
 
 /**
  * Validate pointset related functionality.
  */
-public class PointsetSequences extends SequenceBase {
+public class PointsetSequences extends PointsetBase {
 
-  private static String NO_POINTSET_CONFIG = "no pointset declared in config";
+  public static final String EXTRANEOUS_POINT = "extraneous_point";
+  private static final int DEFAULT_SAMPLE_RATE_SEC = 10;
+
+  private boolean isErrorState(PointPointsetState pointState) {
+    return ofNullable(catchToNull(() -> pointState.status.level)).orElse(Level.INFO.value())
+        >= Level.ERROR.value();
+  }
+
+  private void untilPointsetSanity() {
+    untilTrue("pointset state reports same points as defined in config", () ->
+        deviceState.pointset.points.keySet().equals(deviceConfig.pointset.points.keySet()));
+    untilTrue("pointset event contains correct points with present_value",
+        () -> {
+          List<PointsetEvent> pointsetEvents = popReceivedEvents(PointsetEvent.class);
+          return pointsetEvents.get(pointsetEvents.size() - 1).points.entrySet().stream()
+              .filter(this::validPointEntry).map(Entry::getKey).collect(Collectors.toSet())
+              .equals(deviceConfig.pointset.points.keySet());
+        }
+    );
+  }
+
+  private boolean validPointEntry(Entry<String, PointPointsetEvent> point) {
+    PointPointsetState pointState = deviceState.pointset.points.get(point.getKey());
+    return point.getValue().present_value != null || isErrorState(pointState);
+  }
+
+  @Test(timeout = ONE_MINUTE_MS)
+  @Summary("pointset configuration contains extraneous point")
+  @Feature(stage = ALPHA, bucket = POINTSET)
+  public void pointset_request_extraneous() {
+
+    untilHasInterestingSystemStatus(false);
+    untilPointsetSanity();
+
+    deviceConfig.pointset.points.put(EXTRANEOUS_POINT, new PointPointsetConfig());
+
+    try {
+      untilTrue("pointset status contains extraneous point error",
+          () -> ifNotNullGet(deviceState.pointset.points.get(EXTRANEOUS_POINT),
+              state -> state.status.category.equals(POINTSET_POINT_INVALID)
+                  && state.status.level.equals(POINTSET_POINT_INVALID_VALUE)));
+      untilHasInterestingSystemStatus(true);
```

Comment:
Why is this necessary if the device has already reported the status of the point as an error?

I do have a concern with the system status  checking in sequencer tests generally that devices may correctly, or incorrectly, use that status to relay some information about the device which may prevail over these errors.